### PR TITLE
Refactor/move execution trace to module

### PR DESCRIPTION
**Pull request by iamyulong**

### CODE REVIEW

File: radix-engine/src/engine/modules/execution_trace.rs [hunk 1]
```diff
@@ -18,41 +18,89 @@ pub struct ExecutionTraceReceipt {
...
+        _call_frames: &mut Vec<CallFrame>,
+        _vault_id: VaultId,
+        fee: Resource,
+        _contingent: bool,
+    ) -> Result<Resource, ModuleError> {
+        Ok(fee)
+    }
 }
 
 impl ExecutionTrace {
```
Comment (LukasGasior1) [inline radix-engine/src/engine/modules/execution_trace.rs:85]: now that it's a full-fledged module, should we rename it to ExecutionTraceModule?

File: radix-engine/src/engine/modules/execution_trace.rs [hunk 2]
```diff
@@ -62,157 +110,194 @@ impl ExecutionTrace {
...
+
+    pub fn new(
+        ops: Vec<(REActor, VaultId, VaultOp)>,
+        actual_fee_payments: HashMap<VaultId, Decimal>,
+        state_track: &mut AppStateTrack,
+        is_commit_success: bool,
+    ) -> Self {
+        let mut vault_changes = HashMap::<ComponentId, HashMap<VaultId, Decimal>>::new();
+        let mut vault_locked_by = HashMap::<VaultId, ComponentId>::new();
+        for op in ops {
```
Comment (LukasGasior1) [inline radix-engine/src/engine/modules/execution_trace.rs:221]: any reason not to name the tuple components here? (`for (actor, vault_id, vault_op) in ops`)
not a big fan of using indices

File: radix-engine/src/engine/modules/execution_trace.rs [hunk 3]
```diff
@@ -62,157 +110,194 @@ impl ExecutionTrace {
...
+
+impl ExecutionTraceReceipt {
+    // TODO: is it better to derive resource changes from substate diff, instead of execution trace?
+    // The current approach relies on various runtime invariants.
+
+    pub fn new(
+        ops: Vec<(REActor, VaultId, VaultOp)>,
+        actual_fee_payments: HashMap<VaultId, Decimal>,
+        state_track: &mut AppStateTrack,
+        is_commit_success: bool,
```
Comment (LukasGasior1) [inline radix-engine/src/engine/modules/execution_trace.rs:217]: Does this correspond to `TransactionResult Commit/Reject` or `TransactionOutcome Success/Failure`? Should we use one of those names here for consistency?

**Review by LukasGasior1 — APPROVED** [review]
Just a few minor comments; Nice refactoring :+1: LGTM

File: radix-engine/src/engine/modules/execution_trace.rs (hunk 1 above)
Comment (iamyulong) [reply radix-engine/src/engine/modules/execution_trace.rs:85]: Yes, will do.

File: radix-engine/src/engine/modules/execution_trace.rs (hunk 2 above)
Comment (iamyulong) [reply radix-engine/src/engine/modules/execution_trace.rs:221]: Good idea. That's definitely better!

File: radix-engine/src/engine/modules/execution_trace.rs (hunk 3 above)
Comment (iamyulong) [reply radix-engine/src/engine/modules/execution_trace.rs:217]: Yes, but I think it's consistent regards to naming:
- `Commit`
   * `Success`
   * `Failure`
- `Reject`

Comment (LukasGasior1) [reply radix-engine/src/engine/modules/execution_trace.rs:217]: Ahh, right `is_Commit->Success`, makes sense

**Review by talekhinezh — APPROVED** [review]

**Pull request merged**